web/channels/player_channel.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Yggdrasil.PlayerChannel do
+  use Yggdrasil.Web, :channel
+  alias Yggdrasil.Player
+  alias Yggdrasil.Endpoint
+
+  def join("player:" <> user_id = topic, _message, socket) do
+    if socket.assigns.user == String.to_integer(user_id) do
+      socket = assign socket, :player_topic, topic
+      {:ok, socket}
+    else
+      {:error, %{ error: "auth failure"} }
+    end
+  end
+
+  def handle_in("join_game", _message, socket) do
+    player_topic = socket.assigns.player_topic
+    push_msg = fn (payload) ->
+      Endpoint.broadcast! player_topic, "event", payload
+    end
+
+    case Player.join_game socket.assigns.user, self, push_msg do
```

Comment:
Yeah I think I'm going to leave it as is since there are basically no answered questions as to what the design will be. I'll try to keep chugging away at what I have but we should both assume at some point there will be a lot of reworking as things get fleshed out.